File: 3258. Count Substrings That Satisfy K-Constraint I.java
class Solution {
    public int countKConstraintSubstrings(String s, int k) {
        int answer = 0;
        for (int i = 0; i < s.length(); i++) {
            int count = 0;
            for (int j = i; j < s.length(); j++) {
                if (s.charAt(j) == '1') count++;
                if (count <= k || j-i+1-count <= k) answer++;
            }
        }
        return answer;
    }
}

File: 3258. Count Substrings That Satisfy K-Constraint I.go
func countKConstraintSubstrings(s string, k int) int {
    answer := 0
    for i := 0; i < len(s); i++ {
        count := 0
        for j := i; j < len(s); j++ {
            if s[j] == '1' { count++ }
            if count <= k || j-i+1-count <= k { answer++ }
        }
    }
    return answer
}

